Snippets/clean-nametable.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+"""Script to remove redundant font name table records."""
+from fontTools.ttLib.ttVisitor import TTVisitor
+from fontTools import ttLib
+from fontTools.ttLib import TTFont
+import fontTools.ttLib.tables.otTables as otTables
+from fontTools.ttLib.tables import C_P_A_L_
+import sys
+import logging
+
+
+logger = logging.getLogger()
+
+
+class NameRecordVisitor(TTVisitor):
+    def __init__(self):
+        self.seen = set()
+
+    def removeUnusedNameRecords(self, font):
```

Comment:
Maybe remove this from the visitor object, move it somewhere else.

Reply:
Thanks for looking. I've moved and converted the function so it's a static method of the name table. Imo it feels like belongs here. Happy to drop the static method and just have it as a plain func instead. I've moved it away from the Snippets dir so it can be imported into other libs and tools.

Happy to add tests if you're ok with the above. Still not too sure if I've found all the places in a font which can have a nameID though.